nodemapper/src/gui/Builder/components/HighlightedJSON.tsx

```diff
@@ -34,14 +66,130 @@ const addQuotesIfString = (value: string, isString: boolean) => {
 };
 
 interface HighlightedJSONProps {
+  nodeid: string;
   json: string;
 }
 
+export const lookupKey = (json, keylist: string[], key: string) => {
+  // If key is the empty string, return the last key in the keylist
+  if (key === "") {
+    key = keylist[keylist.length - 1];
+    keylist = keylist.slice(0, keylist.length - 1);
+  }
+  for (let i = 0; i < keylist.length; i++) {
+    if (json[keylist[i]] === undefined) return undefined;
+    json = json[keylist[i]];
+  }
+  if (json[key] === undefined) return undefined;
+  return json[key];
+};
+
+const lookupKeyGlobal = (
+  json,
+  node_name: string,
+  keylist: string[],
+  key: string,
+  lookup_count = 0,
+) => {
+  // Stop infinite recursion
+  if (lookup_count > 100) {
+    return undefined;
+  }
+
+  // If key is the empty string, return the last key in the keylist
+  if (key === "") {
+    if (keylist.length === 0) {
+      return undefined;
+    }
+    key = keylist[keylist.length - 1];
+    keylist = keylist.slice(0, keylist.length - 1);
+  }
```

Comment:
I think this blocked can be moved up as it can just return `undefined`. Or am I missing something?

Reply:
This is a convenience check which allows calling functions to pass only a `keylist` (without a key), as the required index is often already available in a list, and would otherwise need to be separated before each call. Both `key` and `keylist` are used later in the function to return the indexed value, and this code block is already very near the top of the `lookupKeyGlobal` function, so I have not changed this.